app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,50 @@
+class CommentsController < ApplicationController
+  before_action :current_comment, only: %i[edit update destroy]
+
+  def user_comments
+    @comments = @current_user.comments
+  end
+
+  def event_comments
+    @comments = Comment.where(event_id: params[:eventid])
+  end
+
+  def new
+    @event = Event.find(params[:eventid])
```

Comment:
Please create one action for finding event_id and apply filter action for `new and event_comments` and optimize event_comments mathod